tests/0148-offset_fetch_commit_error_mock.c

```diff
@@ -0,0 +1,548 @@
+                rd_kafka_mock_broker_push_request_error_rtts(
+                    mcluster, 1, RD_KAFKAP_OffsetCommit, 1,
+                    RD_KAFKA_RESP_ERR_UNKNOWN_MEMBER_ID,
+                    session_timeout_ms + heartbeat_interval_ms);
+        }
+
+        /* Otherwise partition is committed before leaving the group */
+        test_consumer_close(consumer);
+        rd_kafka_destroy(consumer);
+
+
+        /* Reset mock assignor to automatic */
+        rd_kafka_mock_cgrp_consumer_target_assignment(mcluster, topic, NULL);
+
+        consumer = test_create_consumer(topic, NULL, conf, NULL);
+        test_consumer_subscribe(consumer, topic);
+
+        if (session_times_out) {
+                /* Messages are consumed again because the commit failed */
+                test_msgver_init(&mv, testid);
+                test_consumer_poll_exact("messages consumed again", consumer,
+                                         testid, -1, 0, msgcnt, rd_true, &mv);
+                test_msgver_verify("messages consumed again", &mv,
+                                   TEST_MSGVER_PER_PART, 0, msgcnt);
+                test_msgver_clear(&mv);
+        } else {
+                /* No message should be consumer after the autocommit */
+                test_consumer_poll_no_msgs("no messages", consumer, testid,
+                                           200);
+        }
+        test_consumer_close(consumer);
+        rd_kafka_destroy(consumer);
+        rd_kafka_mock_clear_request_errors(mcluster, RD_KAFKAP_OffsetCommit);
+
+        SUB_TEST_PASS();
+}
+
+static void log_cb_closing_issue(const rd_kafka_t *rk,
+                                 int level,
+                                 const char *fac,
+                                 const char *buf) {
+        if (strstr(buf, "Acknowledging target assignment")) {
+                mtx_lock(&log_lock);
+                ack_target_assignment_count++;
+                if (ack_target_assignment_count == 2) {
+                        trigger_consumer_close = rd_true;
+                        cnd_signal(&log_cnd);
+                }
+                mtx_unlock(&log_lock);
+        }
+        if (strstr(buf, "unable to OffsetCommit") &&
+            strstr(buf, "Broker: The member epoch is stale")) {
+                mtx_lock(&log_lock);
+                got_stale_member_epoch_error++;
+                mtx_unlock(&log_lock);
+        }
+}
+
+/**
+ * @brief This test checks that when a consumer acknowledges revocations and
+ * sends a heartbeat just before leaving the group, the heartbeat response may
+ * still be in flight while the leave process begins. In this scenario, the
+ * heartbeat response must update the member epoch rather than being discarded.
+ * Otherwise, subsequent commit requests (which will be required for leaving)
+ * may fail with a stale member epoch error.
+ */
+void do_test_consumer_group_heartbeat_ack_assignment_close(
```

Comment:
to follow the conventions in this test it should be `do_test_ConsumerGroupHeartbeat_ack_assignment_close`

Reply:
I am changing the name to `do_test_consumer_inflight_heartbeat_on_leave`. Which describe the test better.